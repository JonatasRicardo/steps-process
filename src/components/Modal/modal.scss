@import '../../theme.scss';

.modal {
    position: fixed;
    width: 100vw;
    height: 100vh;
    left: 0;
    top: 0;
    z-index: -1;
    opacity: 0;
    display: flex;
    justify-content: space-around;
    @include transition(all .5s);

    &__header{
        position: relative;
        padding: 5px 13px;
        margin-bottom: 30px;
        h1 {
            color: $colorBlack;
            @include font-size(20px);
            margin: 5px 0 2px;
            font-weight: 500;
            letter-spacing: .3px;
        }
        label {
            @include font-size(12px);
            background-color: $colorRed;
            color: $colorWhite;
            padding: 1px 8px 1px 9px;
            border-radius: 2px;
            font-weight: 100;
            margin: 6px 4px;
            vertical-align: top;
            display: inline-block;
            letter-spacing: 0.9px;
        }
        h4 {
            font-weight: 100;
            margin: 0;
            @include font-size(13px);
            letter-spacing: 0.5px;
        }
        button {
            position: absolute;
            right: 14px;
            top: 11px;
            border: 0;
            background: transparent;
            cursor: pointer;
            border-radius: 50%;
            background-color: $colorWhite;
            padding: 5px;
            text-align: center;
            width: 30px;
            height: 30px;
            @include transition(all .5s);

            img {
                width: 12px;
                height: 12px;
            }

            &:hover {
                box-shadow: 0px 0px 16px -6px rgba(0, 0, 0, 0.75);
            }
            &:active {
                box-shadow: inset 0px 0px 11px -6px rgba(0, 0, 0, 0.75);
            }
        }
    }

    &__body {
        width: 920px;
        max-width: 95%;
        height: 327px;
        background-color: $colorWhite;
        margin-top: 100px;
        z-index: 1;
        border-radius: 5px;
        box-shadow: 10px 10px 68px -17px rgba(0,0,0,0.75);
        position: relative;
        padding-top: 4px;
        margin-top: 200vh;
        
        &:before{
            content: '';
            @include gradient($colorDarkBlue, $colorLightBlue);
            height: 4px;
            width: 100%;
            position: absolute;
            top: 0;
            left: 0;
        }

        &-container{
            border: 1px solid $colorGrayLight;
            border-top: 0;
            padding: 10px;
            border-radius: 0 0 5px 5px;
            background-color: $colorWhite;
        }
    }
    &__overlay{
        background-color: #000;
        opacity: .5;
        position: absolute;
        width: 100%;
        height: 100%;
    }

    &.is-updated{
         @include animation('fadeOut .5s forwards');
    }

    &.is-active {
        @include animation('fadeIn .3s forwards');
        .modal__body{
             @include animation('fromBottom 1s .3s forwards');
        }
    }
}

@include bp-medium{
    .modal{
        &__header{
            padding-right: 50px;
        }
    }
}
@include keyframes(fadeIn) {
    from {
        z-index: -1;
        opacity: 0;
    }
    to {
        z-index: 1;
        opacity: 1;
    }
}

@include keyframes(fadeOut) {
    from {
        z-index: 1;
        opacity: 1;

    }
    to {
        z-index: -1;
        opacity: 0;
    }
}

@include keyframes(fromBottom) {
    from {
        margin-top: 200vh;
    }
    to {
        margin-top: 100px;
    }
}