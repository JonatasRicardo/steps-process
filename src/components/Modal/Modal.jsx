import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import './modal.scss';

const propTypes = {
    active: PropTypes.bool,
}

const defaultProps = {
    active: false,
}

function Modal(props) {
    const { active, children } = props;
    const [didUpdate, setDidUpdate] = useState(false);

    useEffect(() => {
        if(active) setDidUpdate(true);
    }, [active])


    const activeClass = active ? 'is-active' : '';
    const isUpdatedClass = didUpdate ? 'is-updated' : '';
 
    return (
        <div className={`modal ${activeClass} ${isUpdatedClass}`} role="dialog" data-testid="modal">
            <div className="modal__body">
                {children}
            </div>
            <div className="modal__overlay" />
        </div>
    );
}

Modal.propTypes = propTypes;
Modal.defaultProps = defaultProps;

export default Modal;