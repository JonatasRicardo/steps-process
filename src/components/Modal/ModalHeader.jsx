import React from 'react';
import PropTypes from 'prop-types';
import closeIcon from './img/close-icon.png';
import './modal.scss';


const propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    hasCloseButton: PropTypes.bool,
    onCloseButtonClick: PropTypes.func,
    label: PropTypes.string,
}

const defaultProps = {
    hasCloseButton: false,
    onCloseButtonClick: () => {},
}

function ModalHeader(props) {
    const { title, subtitle, hasCloseButton, onCloseButtonClick, label } = props;
 
    return (
        <header className="modal__header">
            <h1>{title} {label && <label>{label}</label>}</h1>
            <h4>{subtitle}</h4>
            {
                hasCloseButton && 
                <button onClick={onCloseButtonClick} data-testid="btn-close-modal">
                    <img src={closeIcon} alt="x" />
                </button>
            }
        </header>
    );
}

ModalHeader.propTypes = propTypes;
ModalHeader.defaultProps = defaultProps;

export default ModalHeader;