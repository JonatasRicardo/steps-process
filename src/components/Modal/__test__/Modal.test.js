import React from 'react';
import { render } from '@testing-library/react';
import Modal from '../Modal';
import ModalHeader from '../ModalHeader';

describe('<Modal/>', () => {
  it('renders Modal components', () => {
    const { getByTestId } = render(
      <Modal />
    );
    const modalElement = getByTestId('modal');
    expect(modalElement).toBeDefined();
  });

  it('Snapshot Modal', () => {
    const { getByTestId } = render(
        <Modal />
    );
    const modalElement = getByTestId('modal');
    expect(modalElement).toMatchSnapshot();
  });

  it('Snapshot Modal with header', () => {
    const { getByTestId } = render(
        <Modal>
          <ModalHeader
              title="Engagement Process"
              subtitle="2019 Fall Oncology Advisory Board"
              label="Abandoned"
              hasCloseButton
          />
        </Modal>
    );
    const modalElement = getByTestId('modal');
    expect(modalElement).toMatchSnapshot();
  });
})
