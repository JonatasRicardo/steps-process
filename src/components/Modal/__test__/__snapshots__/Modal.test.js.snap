// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal/> Snapshot Modal 1`] = `
<div
  class="modal  "
  data-testid="modal"
  role="dialog"
>
  <div
    class="modal__body"
  />
  <div
    class="modal__overlay"
  />
</div>
`;

exports[`<Modal/> Snapshot Modal with header 1`] = `
<div
  class="modal  "
  data-testid="modal"
  role="dialog"
>
  <div
    class="modal__body"
  >
    <header
      class="modal__header"
    >
      <h1>
        Engagement Process
         
        <label>
          Abandoned
        </label>
      </h1>
      <h4>
        2019 Fall Oncology Advisory Board
      </h4>
      <button
        data-testid="btn-close-modal"
      >
        x
      </button>
    </header>
  </div>
  <div
    class="modal__overlay"
  />
</div>
`;
